src/components/Dialogs/GeneralDialog/GeneralDialog.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect } from 'react'
+import { Portal } from '@/components'
+import { useOverlayManager } from '@/hooks/useOverlayManager'
+import { CSSTransition } from 'react-transition-group'
+import { StyledBackdrop, StyledContainer, StyledExitButton, dialogTransitions } from './GeneralDialog.style'
+import { Icon } from '@/shared/components'
+
+export type DialogProps = {
+  showDialog?: boolean
+  exitButton?: boolean
+  handleExit?: (e: React.MouseEvent<HTMLButtonElement>) => void
+}
+
+const GeneralDialog: React.FC<DialogProps> = ({ children, showDialog, exitButton, handleExit }) => {
+  const { overlayContainerRef, handleOverlayOpen, handleOverlayClose } = useOverlayManager()
+
+  useEffect(() => {
+    if (!showDialog) {
+      return
+    }
+    handleOverlayOpen()
+    return () => {
+      handleOverlayClose()
+    }
+  }, [handleOverlayClose, handleOverlayOpen, showDialog])
+  return (
+    overlayContainerRef.current && (
+      <Portal portal={overlayContainerRef}>
+        <CSSTransition in={showDialog} timeout={250} classNames="backdrop" unmountOnExit mountOnEnter>
+          <StyledBackdrop css={dialogTransitions} className="backdrop">
```

Comment:
I know we may have discussed something else, but thinking about it now I don't think the dialog should be a children of the backdrop, doesn't seem to make much semantic sense. Let's make them siblings